sql/src/main/java/io/crate/analyze/relations/RelationSplitter.java

```diff
@@ -110,6 +111,18 @@ private void processOutputs() {
             });
         }
 
+        Optional<List<Symbol>> groupBy = querySpec.groupBy();
+        if (groupBy.isPresent()) {
+            FieldsVisitor.visitFields(groupBy.get(), addFieldToMap);
+        }
+        Optional<HavingClause> having = querySpec.having();
```

Comment:
I could probably do this in a separate PR upfront and create unit tests - I think this wasn't required so far because of how the fetch-rewrite worked (It didn't have strict relation boundaries as we have now that the MSS sources are a QueriedRelation) 

(This is about adding GROUP-KEYS / HAVING expressions which aren't in the outputs to the output of the newly created sub-relations)